### PR TITLE
Regenerate session for anonymous user after succesfully auth

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
Regenerate session after entering the password of a password protected public share for anonymous users 

## Related Issue

- Fixes https://github.com/owncloud/enterprise/issues/4535

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: apps/files_sharing/lib/Controllers/ShareController.php [hunk 1]
```diff
@@ -164,6 +170,9 @@ public function authenticate($token, $password = '') {
 		$authenticate = $this->linkShareAuth($share, $password);
 
 		if ($authenticate === true) {
+			if (!$this->userSession->isLoggedIn()) {
```
Comment (mrow4a) [inline apps/files_sharing/lib/Controllers/ShareController.php:173]: this is to detect public link?

**Review by mrow4a — COMMENTED** [review]
what if I was in normal folder, and then accessed public link? would I need to authenticate again?

File: apps/files_sharing/lib/Controllers/ShareController.php (hunk 1 above)
Comment (JanAckermann) [reply apps/files_sharing/lib/Controllers/ShareController.php:173]: We Check If you was logged in before, so I guess nope

Comment (JanAckermann) [reply apps/files_sharing/lib/Controllers/ShareController.php:173]: No, we don't create a new session for a already logged in user

**Comment by JanAckermann** [comment]
> what if I was in normal folder, and then accessed public link? would I need to authenticate again?

We Check If you was logged in before, so I guess nope

**Comment by JanAckermann** [comment]
> what if I was in normal folder, and then accessed public link? would I need to authenticate again?

No, we only reset the cookie/session for a non-authenticated user, if the cookie was injected before to the authenticated user via XSS, the potential attacker would already have access.

File: apps/files_sharing/lib/Controllers/ShareController.php (hunk 1 above)
Comment (JanAckermann) [reply apps/files_sharing/lib/Controllers/ShareController.php:173]: No, we don't create a new session for a already logged in user, see comment below

**Review by C0rby — APPROVED** [review]
Looks good to me! :+1:

**Pull request merged**